### PR TITLE
T5 Prompt Learning Fixes for Pipeline Parallel

### PR DESCRIPTION
**Pull request by MaximumEntropy**
Signed-off-by: MaximumEntropy <sandeep.subramanian.1@umontreal.ca>

# What does this PR do ?

Fixes hanging with T5 prompt learning for PP > 1

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_t5_prompt_learning_model.py [hunk 1]
```diff
@@ -73,6 +74,7 @@ class MegatronT5PromptLearningModel(MegatronBasePromptLearningModel):
 
     def __init__(self, cfg: DictConfig, trainer: Trainer):
         super().__init__(cfg, trainer)
+        self.model_type = ModelType.encoder_and_decoder
```
Comment (arendu) [inline nemo/collections/nlp/models/language_modeling/megatron_t5_prompt_learning_model.py:77]: is this `self.model_type` used?

Comment (vadam5) [reply nemo/collections/nlp/models/language_modeling/megatron_t5_prompt_learning_model.py:77]: Apex accesses this model property inside it's fwd/bwd func. I think all encoder-decoder models need this, but the default works fine for encoder only or decoder only models. Its needed to know if tensor shapes might be different between encoder and decoder communications when PP > 1 (PP > 2?)

**Review by arendu — APPROVED** [review]
LGTM!

**Review by vadam5 — APPROVED** [review]
@MaximumEntropy said it looks good

**Pull request merged**